src/electron/services/system-info.ts

```diff
@@ -0,0 +1,30 @@
+import { ipcMain } from 'electron'
+import * as os from 'os'
+
+import type { BasicSystemInfo } from '@/types/platform'
+
+/**
+ * Setup system info service
+ * Exposes IPC handler for getting system information including platform and architecture
+ */
+export const setupSystemInfoService = (): void => {
+  ipcMain.handle('get-system-info', (): BasicSystemInfo => {
+    return {
+      platform: os.platform(),
+      arch: os.arch(),
+      processArch: process.arch,
```

Comment:
Should this make use of `getSystemInfo`, instead of re-implementing it?

Reply:
I'm going to have to talk to claudinho about that...